backend/Posts/tests.py

```diff
@@ -20,10 +20,123 @@ def setUp(self):
+        self.edit_user_client.force_authenticate(user=self.edit_user)
+
+        self.edit_hub_owner = User.objects.create_user(username='edit_hub_owner', password='password')
+        self.edit_hub_owner_client = APIClient()
+        self.edit_hub_owner_client.force_authenticate(user=self.edit_hub_owner)
+
+        self.edit_hub_mod = User.objects.create_user(username='edit_hub_mod', password='password')
+        self.edit_hub_mod_client = APIClient()
+        self.edit_hub_mod_client.force_authenticate(user=self.edit_hub_mod)
+
+        self.edit_hub = Hub.objects.create(name='edit_hub', description='A hub to test editing', owner=self.edit_hub_owner)
+        self.edit_hub.mods.add(self.edit_hub_mod)
+        self.edit_hub.members.add(self.edit_user, self.edit_hub_mod)
+
+        self.edit_dummy_post = {'title': "edit post title user", 'message': "edit post message user", 'hub_id': self.edit_hub.id}
+
+
         self.dummy_post = {'title': "DUMMY POST", 'message': "THIS IS A DUMMY POST"} #MUST ADD "hub_id" KEY
         self.dummy_post2 = {'title': "DUMMY POST 2", 'message': "THIS IS A SECOND DUMMY POST"} # MUST ADD 'hub_id' key
         self.factory = APIRequestFactory()
         
+    def test_user_edit_their_post(self):
+        """
+        Make sure a user can edit their own posts.
+        """
+        response = self.edit_user_client.post(reverse('post-create'), self.edit_dummy_post, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED, "Post was not created")
+
+        new_data = {'title': "edited title", 'message': "edited message"}
+
+        response = self.edit_user_client.patch(reverse('post-edit', kwargs={'id': response.data['id']}), new_data, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK, "Post could not be edited by post author")
+        self.assertEqual(response.data['title'], new_data['title'], "Post title did not change")
+        self.assertEqual(response.data['message'], new_data['message'], "Post message did not change")
+    
+    def test_mod_cannot_edit_hub_post(self):
+        """
+        Make sure that a mod cannot edit posts on a hub that they moderate.
+        """
+        response = self.edit_user_client.post(reverse('post-create'), self.edit_dummy_post, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED, "Post was not created")
+        post = response.data
+
+        new_data = {'title': "edited title", 'message': "edited message"}
+
+        response = self.edit_hub_mod_client.patch(reverse('post-edit', kwargs={'id': response.data['id']}), new_data, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN, "Post could be edited by hub mod")
+        self.assertNotEqual(post['title'], new_data['title'], "Post title changed")
+        self.assertNotEqual(post['message'], new_data['message'], "Post message changed")
+    
+    def test_owner_edit_hub_post(self):
```

Comment:
is this still a valid test case? looks like you updated some assertions but idk if it's what you intended based on the comments